pjsip/src/pjsua-lib/pjsua_core.c

```diff
@@ -3735,12 +3735,20 @@ static pj_status_t restart_listener(pjsua_transport_id id,
     switch (tp_info.type) {
     case PJSIP_TRANSPORT_UDP:
     case PJSIP_TRANSPORT_UDP6:
-	status = pjsip_udp_transport_restart2(
-				       pjsua_var.tpdata[id].data.tp,
-				       PJSIP_UDP_TRANSPORT_DESTROY_SOCKET,
-				       PJ_INVALID_SOCKET,
-				       &bind_addr,
-				       NULL);
+    {
+	pjsip_udp_transport_res_param param;
+
+	pjsip_udp_transport_res_param_default(&param);
+	param.option = PJSIP_UDP_TRANSPORT_DESTROY_SOCKET;
+	param.sock = PJ_INVALID_SOCKET;
+	param.local = &bind_addr;
+	param.unlock = &PJSUA_UNLOCK;
+	param.lock = &PJSUA_LOCK;
+
+	status = pjsip_udp_transport_restart3(
+					    pjsua_var.tpdata[id].data.tp,
+					    &param);
+    }
```

Comment:
Why introduce new APIs, instead of simply unlocking PJSUA lock before invoking `pjsip_udp_transport_restart*()` and re-locking later after?